test/unit/math/prim/scal/err/is_size_match_test.cpp

```diff
@@ -0,0 +1,23 @@
+#include <stan/math/prim/scal.hpp>
+#include <gtest/gtest.h>
+#include <test/unit/util.hpp>
+
+TEST(ErrorHandlingMatrix, checkSizeMatch) {
+  using stan::math::is_size_match;
+  int x;
+  size_t y;
+
+  x = 3;
+  y = 4;
+  EXPECT_FALSE(is_size_match(x, y));
+  EXPECT_FALSE(is_size_match(x, y));
```

Comment:
We can remove lines 13-15... they're just duplicates.

Reply:
Maybe what we should be doing is:
```
is_size_match(x, y);
is_size_match(y, x);
is_size_match(x, x);
is_size_match(y, y);
```

That would be to flex the types. If that were the case, I'd maybe rename the variables so it's clear that we're testing types. Or you can use a string afterwards: so... something like
```
EXPECT_FALSE(is_size_match(x, y)) << "int, size_t";
```

Reply:
Oh yeah.. it was late. I think I have what you are looking for.